sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -49,8 +49,10 @@ trait FunctionRegistryBase[T] {
 
   final def registerFunction(
       name: FunctionIdentifier, builder: FunctionBuilder, source: String): Unit = {
+    var builderClassName = builder.getClass.getCanonicalName
```

Comment:
no biggie but I would add a single comment that we're doing this for anonymous class name returning `null` in jdk 17 (with JIRA ID)

Reply:
added